lang/cs/Org.Apache.REEF.Client/YARN/YarnJobCommandBuilder.cs

```diff
@@ -0,0 +1,107 @@
+﻿// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+// 
+//   http://www.apache.org/licenses/LICENSE-2.0
+// 
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+using System.Collections.Generic;
+using System.Text;
+using Org.Apache.REEF.Client.YARN.Parameters;
+using Org.Apache.REEF.Common.Files;
+using Org.Apache.REEF.Tang.Annotations;
+
+namespace Org.Apache.REEF.Client.YARN
+{
+    /// <summary>
+    /// YarnJobCommandBuilder is .NET implementation of `org.apache.reef.runtime.common.launch.JavaLaunchCommandBuilder`
+    /// This class provides the command to be submitted to RM for execution of .NET driver.
+    /// </summary>
+    public sealed class YarnJobCommandBuilder : IYarnJobCommandBuilder
+    {
+        private static readonly string JavaExe = @"%JAVA_HOME%/bin/java";
+
+        // keeping static values for now; make dynamic when needed
+        private static readonly string JvmOptions = @"-XX:PermSize=128m -XX:MaxPermSize=128m -Xmx512m";
```

Comment:
I would prefer it to be configurable. ASA ran out of Java heap space once, and it was most likely caused by a memory leak somewhere in REEF. In that scenario, I don't think we want to compromise our usability if we can just allow the user to increase the heap size.